spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/validation/ValidationAutoConfigurationTests.java

```diff
@@ -207,15 +208,17 @@ void validationCanBeConfiguredToUseJdkProxy() {
 	}
 
 	@Test
+	@SuppressWarnings("unchecked")
 	void userDefinedMethodValidationPostProcessorTakesPrecedence() {
 		this.contextRunner.withUserConfiguration(SampleConfiguration.class).run((context) -> {
 			assertThat(context.getBeansOfType(Validator.class)).hasSize(1);
 			Object userMethodValidationPostProcessor = context.getBean("testMethodValidationPostProcessor");
 			assertThat(context.getBean(MethodValidationPostProcessor.class))
 					.isSameAs(userMethodValidationPostProcessor);
 			assertThat(context.getBeansOfType(MethodValidationPostProcessor.class)).hasSize(1);
-			assertThat(context.getBean(Validator.class))
-					.isNotSameAs(ReflectionTestUtils.getField(userMethodValidationPostProcessor, "validator"));
+			Object validator = ReflectionTestUtils.getField(userMethodValidationPostProcessor, "validator");
```

Comment:
There's actually a bug here. It uses `isNotSameAs` on the validator field. this field is of type `Supplier<Validator>`, so the assert is always true.